spec/01-unit/30-new-dns-client/02-old_client_spec.lua

```diff
@@ -0,0 +1,1516 @@
+    client = nil
+
+    receive_func = nil
+  end)
+
+
+  describe("initialization", function()
+
+    it("succeeds if hosts/resolv.conf fails", function()
+      local cli, err = client.new({
+          nameservers = TEST_NSS,
+          hosts = "non/existent/file",
+          resolv_conf = "non/exitent/file",
+      })
+      assert.is.Nil(err)
+      assert.same(cli.r_opts.nameservers, TEST_NSS)
+    end)
+
+    describe("inject localhost", function()
+
+      it("if absent", function()
+        writefile(resolv_path, "")
+        writefile(hosts_path, "") -- empty hosts
+
+        local cli = assert(client_new())
+        local answers = cli.cache:get("localhost:28")
+        assert.equal("[::1]", answers[1].address)
+
+        answers = cli.cache:get("localhost:1")
+        assert.equal("127.0.0.1", answers[1].address)
+
+        answers = cli:resolve("localhost")
+        assert.equal("127.0.0.1", answers[1].address)
+      end)
+
+      it("not if ipv4 exists", function()
+        writefile(hosts_path, "1.2.3.4 localhost")
+        local cli = assert(client_new())
+
+        -- IPv6 is not defined
+        local answers = cli.cache:get("localhost:28")
+        assert.is_nil(answers)
+
+        -- IPv4 is not overwritten
+        answers = cli.cache:get("localhost:1")
+        assert.equal("1.2.3.4", answers[1].address)
+      end)
+
+      it("not if ipv6 exists", function()
+        writefile(hosts_path, "::1:2:3:4 localhost")
+        local cli = assert(client_new())
+
+        -- IPv6 is not overwritten
+        local answers = cli.cache:get("localhost:28")
+        assert.equal("[::1:2:3:4]", answers[1].address)
+
+        -- IPv4 is not defined
+        answers = cli.cache:get("localhost:1")
+        assert.is_nil(answers)
+      end)
+
+      it("cache evication", function()
+        writefile(hosts_path, "::1:2:3:4 localhost")
+        local cli = assert(client_new())
+
+        local answers = cli.cache:get("localhost:28")
+        assert.equal("[::1:2:3:4]", answers[1].address)
+
+        -- evict it
+        cli.cache:delete("localhost:28")
+        answers = cli.cache:get("localhost:28")
+        assert.equal(nil, answers)
+
+        -- resolve and re-insert it into cache
+        answers = cli:resolve("localhost")
+        assert.equal("[::1:2:3:4]", answers[1].address)
+
+        answers = cli.cache:get("localhost:28")
+        assert.equal("[::1:2:3:4]", answers[1].address)
+      end)
+    end)
+  end)
+
+
+  describe("iterating searches", function()
+    local function hook_query_func_get_list()
+      local list = {}
+      query_func = function(self, original_query_func, name, options)
+        table.insert(list, name .. ":" .. options.qtype)
+        return {} -- empty answers
+      end
+      return list
+    end
+
+    describe("without type", function()
+      it("works with a 'search' option", function()
+        writefile(resolv_path, {
+          "nameserver 198.51.100.0",
+          "search one.com two.com",
```

Comment:
I prefer to use `.test` as top-level domain in tests. Just as a safe-guard that we're not hitting a real name.

Reply:
fixed, if it could be modifed to `.test` suffix